service/bfg/bfg.go

```diff
@@ -1198,16 +1275,35 @@ func (s *Server) handleBtcFinalityByKeystonesRequest(ctx context.Context, bfkr *
 	}, nil
 }
 
-func (s *Server) handleL2KeystonesRequest(ctx context.Context, l2kr *bfgapi.L2KeystonesRequest) (any, error) {
-	log.Tracef("handleL2KeystonesRequest")
-	defer log.Tracef("handleL2KeystonesRequest exit")
+func (s *Server) getL2KeystonesCache() []hemi.L2Keystone {
+	log.Tracef("getL2KeystonesCache")
+	defer log.Tracef("getL2KeystonesCache exit")
+
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	if s.l2keystonesCache == nil {
+		return []hemi.L2Keystone{}
+	}
+
+	results := []hemi.L2Keystone{}
+	for _, v := range s.l2keystonesCache {
+		results = append(results, v)
+	}
```

Comment:
I think pre-allocating the `results` slice and using the built-in `copy` function may improve performance here. (suggestion below is untested)
```suggestion
	results := make([]hemi.L2Keystone, len(s.l2keystonesCache))
	copy(results, s.l2keystonesCache)
```

Reply:
will update